staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/validation/validation.go

```diff
@@ -766,6 +806,45 @@ func ValidateCustomResourceColumnDefinition(col *apiextensions.CustomResourceCol
 	return allErrs
 }
 
+func ValidateCustomResourceSelectableFields(schema *structuralschema.Structural, uniqueSelectableFields sets.Set[string], selectableField *apiextensions.SelectableField, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(selectableField.JSONPath) == 0 {
+		allErrs = append(allErrs, field.Required(fldPath.Child("jsonPath"), ""))
+	} else {
+		// Leverage the field path validation originally built for use with CEL features
+		path, foundSchema, err := cel.ValidFieldPath(selectableField.JSONPath, schema, cel.WithFieldPathAllowArrayNotation(false))
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(fldPath, selectableField.JSONPath, fmt.Sprintf("is an invalid path: %v", err)))
+		} else {
+			if path.Root().String() == "metadata" {
+				allErrs = append(allErrs, field.Invalid(fldPath, selectableField.JSONPath, "must not point to fields in metadata"))
+			}
+			if !allowedSelectableFieldSchema(foundSchema) {
+				allErrs = append(allErrs, field.Invalid(fldPath, selectableField.JSONPath, "must point to a field of type string, boolean or integer. Enum string fields and strings with formats are allowed."))
+			}
+			if uniqueSelectableFields.Has(path.String()) {
+				allErrs = append(allErrs, field.Duplicate(fldPath, selectableField.JSONPath))
+			} else {
+				uniqueSelectableFields.Insert(path.String())
```

Comment:
I don't like this side-effect from a validate function.  the uniqueness is known inside the for loop calling this function, let's check that there.

Reply:
SG. I've reorganized the code to avoid the side-effect.